zebra-network/src/peer_set/set/tests/vectors.rs

```diff
@@ -0,0 +1,63 @@
+use futures::FutureExt;
+use tower::{Service, ServiceExt};
+
+use zebra_chain::parameters::{Network, NetworkUpgrade};
+
+use super::{PeerSetBuilder, PeerVersions};
+use crate::{peer::MinimumPeerVersion, protocol::external::types::Version, Request};
+
+#[test]
+fn peer_set_drop() {
+    //
+    let peer_versions = PeerVersions {
+        peer_versions: vec![Version::min_specified_for_upgrade(
+            Network::Mainnet,
+            NetworkUpgrade::Canopy,
+        )],
+    };
+
+    //
+    let runtime = zebra_test::init_async();
+
+    //
+    let (discovered_peers, _handles) = peer_versions.mock_peer_discovery();
+    let (minimum_peer_version, _best_tip_height) =
+        MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+    //
+    runtime.block_on(async move {
+        let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+            .with_discover(discovered_peers)
+            .with_minimum_peer_version(minimum_peer_version.clone())
+            .build();
+
+        // Wait until the peer set is ready
+        let peer_set1 = peer_set.ready().now_or_never();
```

Comment:
`now_or_never` doesn't schedule the task for wakeup, so I'm not sure if it will work in these tests:
https://docs.rs/futures/0.3.17/futures/future/trait.FutureExt.html#method.now_or_never

@jvff what do you think? Should we use `NowOrLater` instead?

```suggestion
        let peer_set1 = peer_set.ready().await;
```

Reply:
I think any of the alternatives is fine, we're always expecting it to be immediately ready and panicking otherwise. That said, my preference would be:

1. `await` is more future-proof and simpler. The only downside is that the test may take longer if we introduce a bug somehow that prevents the `PeerSet` from being immediately ready (and the bug might not get detected). I think this downside should be rare.
2. `NowOrLater` would be the safest option, but it's a little more cumbersome.
3. `now_or_never` is also valid.

For that last two options, I would suggest refactoring the code to not `match`, but to use `now_or_never().expect("peer set service is always ready")` instead :sweat_smile:

Reply:
> `NowOrLater` would be the safest option, but it's a little more cumbersome.

We can turn `NowOrLater` into a trait method like `now_or_never`, that would make it easier to use.

> `now_or_never` is also valid.

`now_or_never` is only valid if you never use the future or anything it checks for the rest of the test.

That's hard to check, so I'd prefer to use one of the other options.